builtin/credential/userpass/path_users.go

```diff
@@ -155,7 +160,15 @@ func (b *backend) userCreateUpdate(req *logical.Request, d *framework.FieldData)
 		userEntry = &UserEntry{}
 	}
 
+	var passwordWasGiven = false
 	if _, ok := d.GetOk("password"); ok {
+		passwordWasGiven = true
+	}
+	if _, ok := d.GetOk("password_hash"); ok {
+		passwordWasGiven = true
+	}
+
+	if passwordWasGiven {
```

Comment:
This set of checks can simply be made `if d.Get("password").(string) != "" || d.Get("password_hash").(string) != "" {`